### DISCUSSION TITLE
Spiky performance in the night club

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#15144] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Those shadowcasting searchlights generate large spikes in lighting draw time
I suspect they somehow sync, mb after some time
Touching doors aroung club also generates those spikes

###### Note: "Debug window" is in spike mode, it shows max value in last few seconds

https://github.com/user-attachments/assets/41686541-57a6-4244-ac8a-8f01c947b458

At first i thought: you could just make them non shadowcasting, but that might ruin the visuals because for some reason flickering non shadowcasting lights appear much darker

https://github.com/user-attachments/assets/a504aafa-ff4b-4239-873a-12facb0065be

An easy compromise solution could be: you give them 160 range so they won't bleed below floor
And then performance fix boys could use "Disable shadow casting lights" option to fix them

### Reproduction steps

Go to club

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Someone is reading my mind (again)
![Screenshot_4](https://github.com/user-attachments/assets/929cebca-ca6b-43e8-9b7c-923351d24ee2)

**Comment by Regalis11** [comment]
I'm afraid I couldn't find a way to repro this in the vanilla game. The searchlights don't move, and don't seem to cause any lighting recalculations whatsoever (and I'm getting a solid 144 FPS at my end in this save). I suspect it might have something to do with the mods you're using, I see that at least the lights on the sub's turrets have been modified from vanilla.

**Reply by SomeRandomNoobKekeke** [reply]
> I see that at least the lights on the sub's turrets have been modified from vanilla

Huh? They shouldn't be :BaroDev:, at least non of my mods do this explicitly :BaroDev:
If you mean that it's dark and creepy then that's not mods, that's reshade :BaroDev:

Anyway, here's example with 0 mods

[Club 2.zip](https://github.com/user-attachments/files/17748232/Club.2.zip)

https://github.com/user-attachments/assets/b5065557-3d8b-4f01-b656-6effe3181600

As you can see here i'm getting a solid 40 FPS at my end in this save
Is it worth saying that i never had fps below 100 in cold caverns in 1.5.9.1? (at least in non-cities)

Perhaps if you have solid 144 FPS then they just never sync
~Wait a sec, i'll make you an ultimate performance testing device...~
Ackshually, adding lags (e.g. spawning multiple stacks of flashlights under your legs) doesn't trigger those spikes
I might need to investigate deeper
But i'm sure it happens in vanilla and i'm not the only one seeing this
![Screenshot_5](https://github.com/user-attachments/assets/361a1011-a45f-4ffa-b27a-91605bf50e08)
![Screenshot_7](https://github.com/user-attachments/assets/81331d2f-767e-4fb9-baa2-ebb0235fe583)

**Reply by Regalis11** [reply]
> Anyway, here's example with 0 mods

That save still has one mod enabled :baromodder:

But anyway, for some reason I got it to happen in this save now. It seems that the recalculations trigger when some door/hatch in the range of the light opens or closes, so I guess I might've just been unlucky in my firsts tests (maybe no-one just happened to open any doors/hatches). Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/bdbe457c8d329d7a71342cf4b9aa631848d216d7 by making the searchlights non-shadow-casting and reducing their range. Fix coming up in the next unstable build.

**Comment by SomeRandomNoobKekeke** [comment]
Also just changing their range from default 2048 to 150 seems to simplify ChekConvexHullsInRange calculations by a lot

**Comment by Regalis11** [comment]
Addressed in the latest unstable build, closing.
